src/core/JsonConfigCommandCenter.ts

```diff
@@ -14,18 +18,115 @@ export class JsonConfigCommandCenter{
+                .then(response => {
+                    console.log("Retrieved Digital Twin List successfully");
+                    const twins = response.data.data?.items;
+                    const twinMap = new Map();
+
+                    for (const twin of twins) {
+                        const twinId = twin.id;
+                        twinMap.set(twinId, twin);
+                    }
+                    digitalTwinConfigMap(twinMap);
+                })
+                .catch(error => {
+                    console.error("Retrieved Digital Twin List failed");
+                    JsonConfigCommandCenter.errorAlert("Save Cancelled!", error);
+                });
+        })();
+    }
+
+    // call with undefind values to create new config
+    public static onJsonConfigSelect = (
+        configId: number,
+        digitalTwinConfigMap: Map<number, unknown> | null,
+        setSelectedTwinId: (configId: number | null) => void,
+        setJsonConfig: (jsonConfig: JsonConfig | null) => void
+    ): void => {
+        if (configId) {
+            const configMap = digitalTwinConfigMap;
+            if (configMap && configMap.size > 0) {
+                const config = configMap.get(configId);
+                if (config) {
+                    setJsonConfig(config as JsonConfig);
+                }
+            }
+            setSelectedTwinId(configId);
+        }
+        else {
+            setJsonConfig({
+                data: {}
+            } as JsonConfig);
+            setSelectedTwinId(null);
+        }
+    }
+
+    public static errorAlert = (message: string, error: string): void => {
+        const errorMsg = `${error}`.split(" | ")[0].split(": ")[1];
+        const errorcode = `${error}`.split(" | ")[1].split(": ")[1];
+
+        alert(`${message}\n${errorMsg}`);
+    }
 
     public static onModeChange(mode: Modes): void {
-        if(JsonConfigCommandCenter.editorInstance) {
+        if (JsonConfigCommandCenter.editorInstance) {
             JsonConfigCommandCenter.editorInstance.setMode(mode);
         }
     }
+
+    public static onUpdate = (): void => {
+        console.warn("Update As function not implemented");
+    }
+
+    public static onDelete = (): void => {
+        console.warn("Delete As function not implemented");
+    }
+
+    public static onSaveAs = (reloadSavedTwin: (configId: number) => void): void => {
+        const currentJson = JsonConfigCommandCenter.currentJson;
+
+        const options: HttpRequestOptions = {
+            data: {
+                "items": [
+                    { "data": currentJson }
+                ]
+            }
+        };
+
+        if (confirm("Do you want to cretate new twin?")) {
+            (async () => {
+                await cogniteClient.post(`${cogniteClient.getBaseUrl()}/api/playground/projects/${cogniteClient.project}/twins`, options,)
```

Comment:
you can keep this functions in another class. like CDFOperations and create these crud operations that return promises. alerts and the user interface specific logic should be in the component.